pkg/controller/kogitoapp/resource/service_monitor_test.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"reflect"
+	"testing"
+
+	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+
+	dockerv10 "github.com/openshift/api/image/docker10"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	discfake "k8s.io/client-go/discovery/fake"
+	clienttesting "k8s.io/client-go/testing"
+)
+
+func TestNewServiceMonitor(t *testing.T) {
+	port := intstr.FromInt(8080)
+
+	type args struct {
+		kogitoApp   *v1alpha1.KogitoApp
+		dockerImage *dockerv10.DockerImage
+		service     *corev1.Service
+		client      *client.Client
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *monv1.ServiceMonitor
+		wantErr bool
+	}{
+		{
+			"TestNewServiceMonitor",
+			args{
+				&v1alpha1.KogitoApp{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+					},
+				},
+				&dockerv10.DockerImage{
+					Config: &dockerv10.DockerConfig{
+						Labels: map[string]string{
+							"prometheus.io/scrape": "true",
+							"prometheus.io/path":   "/ms",
+							"prometheus.io/port":   "8080",
+							"prometheus.io/scheme": "https",
+						},
+					},
+				},
+				&corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"test": "test",
+						},
+					},
+				},
+				&client.Client{
+					ControlCli: fake.NewFakeClient(),
+					Discovery: &discfake.FakeDiscovery{
```

Comment:
I see that you're using this fake client in many places now, wdyt using [this package](https://github.com/kiegroup/kogito-cloud-operator/tree/master/pkg/test) to return a function with this fake client and  also to have the fake client in a `const`, so we avoid repeating ourselves?

Reply:
Moved it to the [test package](https://github.com/kiegroup/kogito-cloud-operator/pull/98/files#diff-eb7b36bde76ac14decc92ec9f8976b75).